multiqc/modules/whatshap/whatshap.py

```diff
@@ -0,0 +1,321 @@
+
+        # Iterate over all files we found
+        for f in self.find_log_files("whatshap/stats", filehandles=True):
+            sample, data = self.parse_whatshap_stats(f)
+            self.whatshap_stats[sample] = data
+            self.add_data_source(f)
+
+        # Filter to strip out ignored sample names
+        self.whatshap_stats = self.ignore_samples(self.whatshap_stats)
+
+        # Raise UserWarning if we didn't find any data
+        if not self.whatshap_stats:
+            raise UserWarning
+
+        # Write parsed report data to a file
+        self.write_data_file(self.whatshap_stats, "multiqc_whatshap_stats")
+
+        # Add whatshap stats to general statistics
+        self.whatshap_add_general_stats()
+
+        # Create bargraph with total phased bp
+        self.add_bargraph_total_phased()
+
+    def parse_whatshap_stats(self, logfile):
+        """Parse WhatsHap stats file"""
+
+        def process_data(data):
+            """
+            Process the data to convert it to the proper format
+
+            In practice, this only involves the following changes:
+                - Convert all numeric values to integers
+                - Replace 'nan' values with zero
+            """
+            for key, value in data.items():
+                # Try to make the value an int
+                try:
+                    data[key] = int(data[key])
+                except ValueError:
+                    pass
+
+                # Replace 'nan' with zero
+                if value == "nan":
+                    data[key] = 0
+
+        file_content = logfile["f"]
+        # This will later be replaced by the sample name from the file, unless
+        # we are parsing an empty file
+        sample = self.clean_s_name(logfile["s_name"], logfile["root"])
+        # Get the header and remove the "#" character from the first field
+        header = next(file_content).strip().split()
+        header[0] = header[0][1:]
+
+        # The results from this log file, a dictionary for every chromosome
+        results = defaultdict(dict)
+        # Parse the lines that have the data
+        for line in file_content:
+            spline = line.strip().split()
+            data = {field: value for field, value in zip(header, spline)}
+            process_data(data)
+            # Remove the sample and chromsome from the data
+            sample = data.pop("sample")
+            chromosome = data.pop("chromosome")
+            # Insert the current line under chromosome
+            results[chromosome] = data
+
+        # If we were parsing an empty file, we return dummy data (basically all
+        # zero) so the sample statistics are consistent, which eases parsing
+        # later on. WhatsHap uses the fictional 'ALL' chromsome to store the
+        # summary for all results, so we re-use that to store all zero's
+        if not results:
+            results = dict()
+            results["ALL"] = defaultdict(int)
```

Comment:
Why not just ignore the log file completely? Usually I would return `False` here, then catch it above and spit out a `log.error()` message and otherwise skip the file.

Reply:
If I recall correctly, Whatshap can actually output a file that only contains the headers as a 'valid' output. If this is the case, we create the summary category 'ALL', and set that to all zero for the values we report (e.g. number of phased variants)

Reply:
Ah ok, so this is not a completely empty file, but rather a file that only contains headers? Ok, then if it's a valid sample output then that makes sense 👍🏻